website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+        self.newCollectionRename = m.prop('');
+        self.dismissModal = function () {
+            $('.modal').modal('hide');
+        };
+        self.currentPage = m.prop(1);
+        self.totalPages = m.prop(1);
+        self.calculateTotalPages = function _calculateTotalPages(result){
+            if(result){ // If this calculation comes after GET call to collections
+                self.totalPages(Math.ceil((result.links.meta.total + args.systemCollections.length)/self.pageSize()));
+            } else {
+                self.totalPages(Math.ceil((self.collections().length)/self.pageSize()));
+            }
+        };
+        self.pageSize = m.prop(5);
+        self.validation = m.prop(false);
+        self.validationError = m.prop('');
+        self.showCollectionMenu = m.prop(false); // Show hide ellipsis menu for collections
+        self.collectionMenuObject = m.prop({item : {label:null}, x : 0, y : 0}); // Collection object to complete actions on menu
+        self.resetCollectionMenu = function () {
+            self.collectionMenuObject({item : {label:null}, x : 0, y : 0});
+        };
+        self.updateCollectionMenu = function _updateCollectionMenu (item, event) {
+            var offset = $(event.target).offset();
+            var x = offset.left;
+            var y = offset.top;
+            if (event.view.innerWidth < 767){
+                x = x-105; // width of this menu plus padding
+                y = y-270; // fixed height from collections parent to top with adjustments for this menu div
+            }
+            self.showCollectionMenu(true);
+            item.renamedLabel = item.label;
+            self.collectionMenuObject({
+                item : item,
+                x : x,
+                y : y
+            });
+        };
+        // Default system collections
+        self.collections = m.prop([
+            new LinkObject('collection', { path : 'users/me/nodes/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors' }, systemCollection : 'nodes'}, 'All My Projects'),
+            new LinkObject('collection', { path : 'users/me/registrations/', query : { 'related_counts' : true, 'page[size]'  : 12, 'embed' : 'contributors'}, systemCollection : 'registrations'}, 'All My Registrations')
+        ]);
+        // Load collection list
+        var loadCollections = function _loadCollections (url){
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            promise.then(function(result){
+                result.data.forEach(function(node){
+                    var count = node.relationships.linked_nodes.links.related.meta.count;
+                    self.collections().push(new LinkObject('collection', { path : 'collections/' + node.id + '/linked_nodes/', query : { 'related_counts' : true, 'embed' : 'contributors' }, systemCollection : false, node : node, count : m.prop(count) }, node.attributes.title));
```

Comment:
As suggested in previous comments, mithril expects props to be updated by invoking them.

```js
var links = result.data.map(...);
self.collections(links);
```